ebill_postfinance/models/ebill_postfinance_service.py

```diff
@@ -0,0 +1,153 @@
+            self.username,
+            self.password,
+            self.biller_id,
+            self.operation_timeout,
+        )
+
+    def test_ping(self):
+        """Test the service from the UI."""
+        self.ensure_one()
+        msg = ["Test connection to service"]
+        res = self.ping_service()
+        if res:
+            msg.append("Success pinging service \n  Receive :{}".format(res))
+        else:
+            msg.append(" - Failed pinging service")
+        raise UserError("\n".join(msg))
+
+    def ping_service(self, test_error=False, test_exception=False):
+        """Ping the service, uses the authentication.
+
+        test_error: will create an unhandled error in the repsonse
+        test_exception: will create a FaultException
+
+        """
+        service = self._get_service()
+        return service.ping()
+
+    def search_invoice(self, transaction_id=None):
+        """Get invoice status from the server.
+
+        transaction_id:
+        """
+        service = self._get_service()
+        res = service.search_invoices(transaction_id)
+        if res.InvoiceCount == 0:
+            _logger.info("Search invoice returned no invoice")
+            return res
+        if res.InvoiceCount < res.TotalInvoiceCount:
+            # TODO handle the case where there is more to download ?
+            _logger.info("Search invoice has more to download")
+        for message in res.InvoiceList.SearchInvoice:
+            _logger.info(f"Found record for message {message}")
+            record = self.invoice_message_ids.search(
+                [("transaction_id", "=", message.TransactionId)]
+            )
+            if record:
+                record.update_message_from_server_data(message)
+            else:
+                _logger.warning(f"Could not find record for message {message}")
+        return res
+
+    def upload_file(self, transaction_id, file_type, data):
+        service = self._get_service()
+        res = service.upload_files(transaction_id, file_type, data)
+        return res
+
+    def get_invoice_list(self, archive_data=False):
+        service = self._get_service()
+        res = service.get_invoice_list(archive_data)
+        return res
+
+    def get_process_protocol_list(self, archive_data=False):
+        # Is this the processing result of an invoice ?
+        service = self._get_service()
+        res = service.get_process_protocol_list(archive_data)
+        return res
+
+    def get_ebill_recipient_subscription_status(self, recipient_id):
+        service = self._get_service()
+        res = service.get_ebill_recipient_subscription_status(recipient_id)
+        return res
+
+    def get_registration_protocol_list(self, archive_data=False):
+        service = self._get_service()
+        res = service.get_registration_protocol_list(archive_data)
+        for registration_protocol in res or {}:
+            self.get_registration_protocol(registration_protocol.CreateDate)
```

Comment:
the value returned from this function is not used

Reply:
This call is related to the contract subscription update through the web service. And not yet implemented, actually when I tested it I could never get any data.
I learn today that the subscription update through the web service is only implemented for b2ebanking or b2c.